packages/wasm-parser/src/decoder.js

```diff
@@ -432,7 +432,7 @@ export function decode(ab: ArrayBuffer, opts: DecoderOpts): Program {
           throw new CompileError(`function signature not found (${typeindex})`);
         }
 
-        const id = t.numberLiteralFromRaw(typeindex);
+        const id = getUniqueName("func");
```

Comment:
Maybe we should try to eliminate the unique name generator (because statefull), and rather use the `typeindex` here.